test/index.test.js

```diff
@@ -102,166 +105,252 @@ describe('AdRoll', function() {
         analytics.stub(window.__adroll, 'record_user');
       });
 
-      it('should track page view with fullName', function() {
-        analytics.page('Category', 'Name', { url: 'http://localhost:34448/test/' });
-        analytics.called(window.__adroll.record_user, {
-        adroll_segments: 'viewed_category_name_page',
-        adroll_custom_data: {
-          path: '/test/',
-          referrer: '',
-          title: 'integrations tests',
-          search: '',
-          name: 'Name',
-          category: 'Category',
-          url: 'http://localhost:34448/test/'
-        }
+      describe('V1', function() {
+        beforeEach(function() {
+          adroll.options.v1 = true;
         });
-      });
 
-      it('should track unnamed/categorized page', function() {
-        analytics.page({ url: 'http://localhost:34448/test/' });
-        analytics.called(window.__adroll.record_user, {
-          adroll_segments: 'loaded_a_page',
-          adroll_custom_data: {
+        it('should track page view with fullName', function() {
+          analytics.page('Category', 'Name', { url: 'http://localhost:34448/test/' });
+          analytics.called(window.__adroll.record_user, {
+            adroll_segments: 'viewed_category_name_page',
             path: '/test/',
             referrer: '',
             title: 'integrations tests',
             search: '',
+            name: 'Name',
+            category: 'Category',
             url: 'http://localhost:34448/test/'
-          }
+          });
+        });
+
+        it('should track unnamed/categorized page', function() {
+          analytics.page({ url: 'http://localhost:34448/test/' });
+          analytics.called(window.__adroll.record_user, {
+            adroll_segments: 'loaded_a_page',
+            path: '/test/',
+            referrer: '',
+            title: 'integrations tests',
+            search: '',
+            url: 'http://localhost:34448/test/'
```

Comment:
Sounds good. I wasn't looking at this update as an opportunity to rewrite the existing integration or its tests, so i just used the style that was there before. 

This update really was originally meant to be an urgent patch to stop sending them useless segment names for new customers (and since i figured it would be quick, I rolled in the other 2 fixes as well). I see now that was probably misguided on multiple fronts (conflating multiple issues into a single PR, ignoring flagrant issues with the existing base, etc :)

I can rewrite the tests, but I do feel strongly that doing so should be out of the immediate scope  and not hinder forward movement on this particular pull; I'm not adding any additional debt to what was already there, and this update has been an absurdly long time coming (which is my own fault) and needs to go out soon!